Source: http://git.savannah.gnu.org/gitweb/?p=gnulib.git;a=history;f=m4/acl.m4;h=03c0f107ce46555aee78e5262bf4bed2f07789ef;hb=HEAD
Upstream: yes, this has been updated in gnulib git. gettext uses their macros in their releases, next release will presumably have updated macros
Reason: gettext uses old gnulib m4 macros, i scrambled together some changes to m4/acl.m4 to make --enable/--disable-acl work

--- gettext-0.17/gettext-tools/gnulib-m4/acl.m4	2007-05-13 18:57:48.000000000 +0200
+++ gettext-0.17/gettext-tools/gnulib-m4/acl.m4	2010-03-04 19:34:56.365301157 +0100
@@ -1,56 +1,70 @@
 # acl.m4 - check for access control list (ACL) primitives
 
-# Copyright (C) 2002, 2004, 2005, 2006, 2007 Free Software Foundation, Inc.
+# Copyright (C) 2002, 2004-2008 Free Software Foundation, Inc.
 # This file is free software; the Free Software Foundation
 # gives unlimited permission to copy and/or distribute it,
 # with or without modifications, as long as this notice is preserved.
 
 # Written by Paul Eggert and Jim Meyering.
 
-AC_DEFUN([AC_FUNC_ACL],
+AC_DEFUN([gl_FUNC_ACL],
 [
   AC_LIBOBJ([acl])
   AC_LIBOBJ([file-has-acl])
 
-  dnl Prerequisites of lib/acl.c.
-  AC_CHECK_HEADERS(sys/acl.h)
-  AC_CHECK_FUNCS(acl)
-  ac_save_LIBS="$LIBS"
-    AC_SEARCH_LIBS(acl_get_file, acl,
-		   [test "$ac_cv_search_acl_get_file" = "none required" ||
-		    LIB_ACL=$ac_cv_search_acl_get_file])
-    AC_SUBST(LIB_ACL)
-    AC_CHECK_HEADERS(acl/libacl.h)
-    AC_CHECK_FUNCS(acl_get_file acl_get_fd acl_set_file acl_set_fd \
-		   acl_free acl_from_mode acl_from_text \
-		   acl_delete_def_file acl_extended_file)
+  AC_ARG_ENABLE([acl],
+    AC_HELP_STRING([--disable-acl], [do not support ACLs]),
+    , [enable_acl=auto])
+
+  LIB_ACL=
+  use_acl=0
+  if test "x$enable_acl" != "xno"; then
+    dnl Prerequisites of lib/acl.c.
+    AC_CHECK_HEADERS(sys/acl.h)
     if test $ac_cv_header_sys_acl_h = yes; then
+      ac_save_LIBS=$LIBS
+      AC_CHECK_FUNCS([acl])
       use_acl=1
-      if test $ac_cv_func_acl_get_file = yes; then
-	# If we detect the acl_get_file bug, disable ACL support altogether.
-	gl_ACL_GET_FILE( , [use_acl=0])
-      fi
-    else
-      use_acl=0
+      AC_SEARCH_LIBS([acl_trivial], [sec],
+	[test "$ac_cv_search_acl_trivial" = "none required" ||
+	 LIB_ACL=$ac_cv_search_acl_trivial
+	 AC_CHECK_FUNCS([acl_trivial])],
+	[AC_CHECK_FUNCS([acl_trivial])
+	 if test $ac_cv_func_acl_trivial != yes; then
+	   AC_SEARCH_LIBS([acl_get_file], [acl],
+	     [test "$ac_cv_search_acl_get_file" = "none required" ||
+	      LIB_ACL=$ac_cv_search_acl_get_file
+	      AC_CHECK_FUNCS(
+		[acl_get_file acl_get_fd acl_set_file acl_set_fd \
+		 acl_free acl_from_mode acl_from_text \
+		 acl_delete_def_file acl_extended_file])
+	      if test $ac_cv_func_acl_get_file = yes; then
+		# If the acl_get_file bug is detected, disable all ACL support.
+		gl_ACL_GET_FILE( , [use_acl=0])
+	      fi
+	      if test $use_acl = 1; then
+		AC_CHECK_HEADERS([acl/libacl.h])
+		if test $ac_cv_func_acl_get_file = yes &&
+		   test $ac_cv_func_acl_free = yes; then
+		  AC_REPLACE_FUNCS([acl_entries])
+		fi
+	      else
+		LIB_ACL=
+	      fi])
+	 fi])
+      LIBS=$ac_save_LIBS
     fi
-    if test $use_acl = 1 &&
-       test $ac_cv_func_acl_get_file = yes &&
-       test $ac_cv_func_acl_free = yes; then
-      AC_REPLACE_FUNCS([acl_entries])
+    if test "x$enable_acl$use_acl" = "xyes0"; then
+      AC_MSG_ERROR([ACLs enabled but support not detected])
     fi
-  LIBS="$ac_save_LIBS"
-  if test $use_acl = 1; then
-    ac_save_LIBS="$LIBS"
-    AC_SEARCH_LIBS([acl_trivial], [sec],
-      [AC_DEFINE([HAVE_ACL_TRIVIAL], 1,
-	 [Define to 1 if you have the `acl_trivial' function.])
-       test "$ac_cv_search_acl_trivial" = "none required" ||
-       LIB_ACL_TRIVIAL="$ac_cv_search_acl_trivial"])
-    AC_SUBST([LIB_ACL_TRIVIAL])
-    LIBS="$ac_save_LIBS"
   fi
-  AC_DEFINE_UNQUOTED(USE_ACL, $use_acl,
-		     [Define if you want access control list support.])
+  AC_SUBST([LIB_ACL])
+  AC_DEFINE_UNQUOTED([USE_ACL], [$use_acl],
+    [Define to nonzero if you want access control list support.])
+
+  # This is for backwards compatibility; remove this by the end of 2007.
+  LIB_ACL_TRIVIAL=
+  AC_SUBST([LIB_ACL_TRIVIAL])
 ])
 
 # gl_ACL_GET_FILE(IF-WORKS, IF-NOT)
--- gettext-0.17/gettext-tools/gnulib-m4/gnulib-comp.m4	2007-10-28 23:54:28.000000000 +0100
+++ gettext-0.17/gettext-tools/gnulib-m4/gnulib-comp.m4	2010-03-04 19:40:54.361479422 +0100
@@ -46,7 +46,7 @@ AC_DEFUN([gl_INIT],
   AM_CONDITIONAL([GL_COND_LIBTOOL], [true])
   gl_cond_libtool=true
   gl_source_base='gnulib-lib'
-  AC_FUNC_ACL
+  gl_FUNC_ACL
   gl_FUNC_ALLOCA
   gl_ARGMATCH
   gl_FUNC_ATEXIT
